database/enums/enum_batched_migration_status.pg

```diff
@@ -0,0 +1 @@
+pending, paused, running, finalizing, failed, succeeded
```

Comment:
Can you help me understand the use of the `running` and `finalizing` states? Are these purely informational or do they actually contain useful information for the state machine?

Reply:
Good question! `running` says "this batched migration is running asynchronously and can safely be worked on by any machine". `finalizing` says "this batched migration is being run synchronously to completion by a specific instance and other machines should not attempt to work on it".

Reply:
How is `running` different than `pending`? It seems that as soon as we are in `pending` state it should be good to go?

What happens if a machine dies while it is doing the finalization step? I would prefer to not have a `finalizing` state (just be in `running`/`pending`) and instead just have the finalizing machine take a lock and then run to completion. I can't see the value in blocking other machines from processing blocks during this time?

Reply:
`pending` represents the fact that batched migrations are technically queued. Say there's a migration `foo` that's half complete and we deploy code that calls `enqueueBatchedMigration('bar')`. `bar` won't immediately start running; it will only start being processed once `foo` completes. This isn't strictly necessary, but I figured it would be useful to specifically define this behavior *and* I think it's generally true that we'll want to spend our resources running older migrations first.

If a machine dies while it's doing finalization, the corresponding non-batch migration won't be finished, so the next instance to try to run that migration will see that the batched migration is already in the `finalizing` state and pick up where the last one left off.

I didn't implement this yet, but the value in "stealing" the work from other machines is that we can then print progress as we go for anyone who is monitoring the migration/deployment. We don't get that if we allow other random hosts to work on the jobs.

Reply:
Ok, I'm convinced! Thanks for the clear explanations!

Reply:
Yay! I just implemented logging during finalization, here's what that will look like:

<img width="601" alt="Screenshot 2023-04-12 at 09 46 58" src="https://user-images.githubusercontent.com/1476544/231526609-a6b6b9d7-8f98-40fa-93ea-c3ca645791d8.png">

Look reasonable?

Reply:
Why do we want the feature of being able to 'finalize' a migration though?

Isn't it good enough to just be able to check if its completed? I don't see why we want to be able to force a machine to synchronously finish a migration.

Reply:
@SethPoulsen is there still an outstanding question here, or did we clear this up in the other thread?